### PR TITLE
Key Vault Parameters

### LINKED ISSUE #43: Update the helium-csharp repo to remove non-language specific instructions

### LINKED ISSUE #58: Pass in KeyVault auth type
- Add a command line param and env var for specific key vault authentication type

- Managed Identity
  - value: MSI
  - default value
  - cmd line: --authtype MSI
  - env var: AUTH_TYPE=MSI

- Azure CLI cached credential
  - value: CLI
  - cmd line: --authtype CLI
  - env var: AUTH_TYPE=CLI

- Visual Studio cached credential
  - value: VS
  - cmd line: --authtype VS
  - env var: AUTH_TYPE=VS
- no auto fallback
- no support for other identity types

- Unused / invalid parameters should be an error with exit(-1) and display Usage()

- add Usage() for no args or -h --help as the first arg.

- Required
  - --kvname - name or full URL of the Azure Key Vault

- Optional
  - -h --help - display command line usage
  - --authtype - authentication type
  - MSI - managed identity (default)
  - CLI - Azure CLI cached credentials
  - VS - Visual Studio cached credentials

### PR DESCRIPTION
**Pull request by bartr**
## Type of PR

- [X] Documentation changes
- [X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [X] GitHub template sync

## Purpose of PR
- add --kvname and --authtype env var and cmd line params

## Issues Closed or Referenced

- Closes #58 
- Closes #43

### CODE REVIEW

**Comment by aflinchb** [comment]
Two things I noticed:
- Need to update the readme with new CLI/environment variable instructions
- I didn't see where (per https://github.com/retaildevcrews/helium-csharp/issues/58), unused/unexpected args result in an error.

**Review by aflinchb — CHANGES_REQUESTED** [review]
Comments (some dupe of above):

- Update readme with new instructions
- Exit w/ usage when unknown argument is provided
- Add -h/ --help

File: README.md [hunk 1]
```diff
@@ -53,7 +53,8 @@ cd src/app
 
 # run in the background
 # $He_Name is set to the name of your key vault
-dotnet run $He_Name &
+# this will use CLI cached credentials
+dotnet run -- --keyvaultname $He_Name --authtype CLI &
```
Comment (aflinchb) [inline README.md:57]: should be --kvname

File: src/app/Program.cs [hunk 2]
```diff
@@ -316,27 +376,100 @@ static async Task<IWebHost> BuildHost(string kvUrl)
...
+            // handle null
+            if (args == null || args.Length == 0 && (kvName == null))
+            {
+                helpFlag = true;
+                return true;
+            }
+
+            // handle -h or --help
+            if (args.Length == 1 && (args[0].ToUpperInvariant() == "-H" || args[0].ToUpperInvariant() == "--help"))
+            {
```
Comment (aflinchb) [inline src/app/Program.cs:387]: should be --HELP

**Review by aflinchb — APPROVED** [review]
Approved w/ comments.

I took a quick look at the failed unit test in the pr github action build, but didn't see it repro locally so not sure what's going on there.

**Review by sanjeevdwivedi — APPROVED** [review]

**Pull request merged**